src/UnitTests/NullBehavior.cs

```diff
@@ -82,6 +84,12 @@ public void Should_return_null_for_nullable_properties()
             }
 
             [Fact]
+            public void Should_return_not_null_for_nullable_properties_that_are_user_defined()
+            {
+                _result.NullableMapFrom.ShouldEqual(0);

```

Comment:
OK then if that's the case you need a MUCH more complicated solution editing the NullExpressionVisitor then.  Just make every conditional option you have not throw an exception.